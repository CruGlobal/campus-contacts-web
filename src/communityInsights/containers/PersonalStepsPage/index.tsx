import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import moment from 'moment';

import Card from '../../components/Card';
import { AppContext } from '../../../appContext';
import StagesSummary from '../../components/StagesSummary';
import StepsChart from '../../components/StepsChart';
import ImpactInfo from '../../components/ImpactInfo';
import FiltersChart from '../../components/FiltersChart';

import {
    GET_IMPACT_REPORT_MOVED,
    GET_IMPACT_REPORT_STEPS_TAKEN,
    GET_STAGES_REPORT_MEMBER_COUNT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT_STEPS_ADDED,
    GET_TOTAL_STEPS_COMPLETED_SUMMARY,
} from './queries';
// GET_IMPACT_REPORT_STEPS_TAKEN INTERFACES
import { impactReportPersonalStepsCompletedCount } from './__generated__/impactReportPersonalStepsCompletedCount';
// GET_TOTAL_STEPS_COMPLETED_SUMMARY INTERFACES
import {
    communityReportStagesPersonalStepsCompleted,
    communityReportStagesPersonalStepsCompleted_community_report_stagesReport as communityReportStagesPersonalStepsCompletedEntry,
} from './__generated__/communityReportStagesPersonalStepsCompleted';
// GET_STEPS_COMPLETED_REPORT INTERFACES
import {
    communityReportDaysPersonalSteps,
    communityReportDaysPersonalSteps_community_report_daysReport as communityReportDaysPersonalRow,
    communityReportDaysPersonalSteps_community_report_daysReport_stageResults as communityReportDaysPersonalStage,
} from './__generated__/communityReportDaysPersonalSteps';
// GET_STAGES_REPORT_STEPS_ADDED INTERFACES
import {
    communityReportStagesPersonalStepsAdded,
    communityReportStagesPersonalStepsAdded_community_report_stagesReport as communityReportStagesPersonalStepsAddedRow,
} from './__generated__/communityReportStagesPersonalStepsAdded';
// GET_IMPACT_REPORT_MOVED INTERFACES
import { impactReportStageProgressionCount } from './__generated__/impactReportStageProgressionCount';
// GET_STAGES_REPORT_MEMBER_COUNT INTERFACES
import {
    communityReportStagesPersonalMemberCount,
    communityReportStagesPersonalMemberCount_community_report_stagesReport as communityReportStagesPersonalMemberCountRow,
} from './__generated__/communityReportStagesPersonalMemberCount';

const PersonalStepsPage = () => {
    const { t } = useTranslation('insights');
    const { orgId } = useContext(AppContext);

    return (
        <div>
            <ImpactInfo
                query={GET_IMPACT_REPORT_STEPS_TAKEN}
                text={(report: impactReportPersonalStepsCompletedCount) =>
                    t('personalSteps.taken', {
                        count:
                            report.community.impactReport
                                .personalStepsCompletedCount,
                    })
                }
                variables={{ id: orgId }}
            />
            <Card title={t('personalSteps.completedTotal')}>
                <StagesSummary
                    query={GET_TOTAL_STEPS_COMPLETED_SUMMARY}
                    variables={{ id: orgId }}
                    mapData={(
                        data: communityReportStagesPersonalStepsCompleted,
                    ) =>
                        data.community.report.stagesReport.map(
                            (
                                entry: communityReportStagesPersonalStepsCompletedEntry,
                            ) => ({
                                stage: entry.stage.name,
                                icon: entry.stage.name
                                    .toLowerCase()
                                    .replace(' ', '-'),
                                count: entry.personalStepsCompletedCount,
                            }),
                        )
                    }
                />
            </Card>
            <Card
                title={t('personalSteps.completed')}
                subtitle={t('personalSteps.completedSubtitle')}
            >
                <FiltersChart
                    query={GET_STEPS_COMPLETED_REPORT}
                    variables={{ id: orgId }}
                    mapData={(data: communityReportDaysPersonalSteps) =>
                        data.community.report.daysReport.map(
                            (row: communityReportDaysPersonalRow) => ({
                                ['total']: row.personalStepsCompletedCount,
                                ['stages']: row.stageResults.map(
                                    (
                                        stage: communityReportDaysPersonalStage,
                                    ) => ({
                                        name: stage.stage.name,
                                        count:
                                            stage.personalStepsCompletedCount,
                                    }),
                                ),
                                ['date']: row.date,
                            }),
                        )
                    }
                    currentDate={moment().toDate()}
                    label={t('personalSteps.legend')}
                />
            </Card>
            <Card
                title={t('personalSteps.added')}
                subtitle={t('personalSteps.addedSubtitle')}
            >
                <StepsChart
                    query={GET_STAGES_REPORT_STEPS_ADDED}
                    mapData={(data: communityReportStagesPersonalStepsAdded) =>
                        data.community.report.stagesReport.map(
                            (
                                row: communityReportStagesPersonalStepsAddedRow,
                            ) => ({
                                [t(
                                    'personalSteps.label',
                                )]: row.personalStepsAddedCount,
                                [t('stage')]: row.stage.name.toUpperCase(),
                            }),
                        )
                    }
                    label={t('personalSteps.label')}
                    index={t('stage')}
                    variables={{
                        period: 'P10Y',
                        id: orgId,
                        endDate: moment().format(),
                    }}
                />
            </Card>

            <ImpactInfo
                query={GET_IMPACT_REPORT_MOVED}
                text={(report: impactReportStageProgressionCount) =>
                    t('personalSteps.reached', {
                        count:
                            report.community.impactReport.stageProgressionCount,
                    })
                }
                variables={{ id: orgId }}
            />
            <Card
                title={t('personalSteps.members')}
                subtitle={t('personalSteps.membersSubtitle')}
            >
                <StepsChart
                    query={GET_STAGES_REPORT_MEMBER_COUNT}
                    mapData={(data: communityReportStagesPersonalMemberCount) =>
                        data.community.report.stagesReport.map(
                            (
                                row: communityReportStagesPersonalMemberCountRow,
                            ) => ({
                                [t('members')]: row.memberCount,
                                [t('stage')]: row.stage.name.toUpperCase(),
                            }),
                        )
                    }
                    label={t('members')}
                    index={t('stage')}
                    variables={{
                        period: 'P10Y',
                        id: orgId,
                        endDate: moment().format(),
                    }}
                />
            </Card>
        </div>
    );
};

export default PersonalStepsPage;
