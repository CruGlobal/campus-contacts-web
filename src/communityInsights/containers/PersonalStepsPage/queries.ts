import gql from 'graphql-tag';

const GET_IMPACT_REPORT_MOVED = gql`
    query impactReportStageProgressionCount($id: ID!) {
        community(id: $id) {
            impactReport {
                stageProgressionCount
            }
        }
    }
`;

const GET_IMPACT_REPORT_STEPS_TAKEN = gql`
    query impactReportPersonalStepsCompletedCount($id: ID!) {
        community(id: $id) {
            impactReport {
                personalStepsCompletedCount
            }
        }
    }
`;

const GET_STAGES_REPORT_MEMBER_COUNT = gql`
    query communityReportStagesPersonalMemberCount(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        memberCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_STAGES_REPORT_STEPS_ADDED = gql`
    query communityReportStagesPersonalStepsAdded(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        personalStepsAddedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_STEPS_COMPLETED_REPORT = gql`
    query communityReportDaysPersonalSteps(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                daysReport {
                    nodes {
                        date
                        personalStepsCompletedCount
                        stageResults {
                            nodes {
                                personalStepsCompletedCount
                                stage {
                                    name
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`;

const GET_TOTAL_STEPS_COMPLETED_SUMMARY = gql`
    query communityReportStagesPersonalStepsCompleted(
        $id: ID!
        $period: String!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        personalStepsCompletedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

export {
    GET_IMPACT_REPORT_MOVED,
    GET_IMPACT_REPORT_STEPS_TAKEN,
    GET_STAGES_REPORT_MEMBER_COUNT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT_STEPS_ADDED,
    GET_TOTAL_STEPS_COMPLETED_SUMMARY,
};
