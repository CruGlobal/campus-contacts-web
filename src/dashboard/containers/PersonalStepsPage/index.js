import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import moment from 'moment';
import _ from 'lodash';

import Card from '../../components/Card';
import AppContext from '../../appContext';
import StagesSummary from '../../components/StagesSummary';
import StepsChart from '../../components/StepsChart';
import ImpactInfo from '../../components/ImpactInfo';
import FiltersChart from '../../components/FiltersChart';

import {
    GET_IMPACT_REPORT_MOVED,
    GET_IMPACT_REPORT_STEPS_TAKEN,
    GET_STAGES_REPORT_MEMBER_COUNT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT_STEPS_ADDED,
    GET_TOTAL_STEPS_COMPLETED_SUMMARY,
} from './queries';

const PersonalStepsPage = () => {
    const { t } = useTranslation('insights');
    const { orgId } = useContext(AppContext);

    return (
        <div>
            <ImpactInfo
                query={GET_IMPACT_REPORT_STEPS_TAKEN}
                text={report =>
                    t('personalSteps.taken', {
                        count: report.impactReport.stepOwnersCount,
                        year: moment().format('YYYY'),
                    })
                }
                variables={{ organizationId: orgId }}
            />
            <Card title={t('personalSteps.completedTotal')}>
                <StagesSummary
                    query={GET_TOTAL_STEPS_COMPLETED_SUMMARY}
                    variables={{
                        organizationId: orgId,
                    }}
                    mapData={data =>
                        data.communityReport.communityStagesReport.map(
                            entry => ({
                                stage: entry.pathwayStage,
                                icon: entry.pathwayStage
                                    .toLowerCase()
                                    .replace(' ', '-'),
                                count: entry.personalStepsCompletedCount,
                            }),
                        )
                    }
                />
            </Card>
            <Card
                title={t('personalSteps.completed')}
                subtitle={t('personalSteps.completedSubtitle')}
            >
                <FiltersChart
                    query={GET_STEPS_COMPLETED_REPORT}
                    mapData={data =>
                        data.communityReport.dayReport.map(row => ({
                            ['total']: row.personalStepsCompletedCount,
                            ['stages']: row.communityStagesReport.map(
                                stage => ({
                                    name: stage.pathwayStage,
                                    count: stage.personalStepsCompletedCount,
                                }),
                            ),
                            ['date']: row.date.substring(0, 2),
                        }))
                    }
                    countAverage={data =>
                        Math.floor(_.sumBy(data, 'total') / data.length)
                    }
                    label={t('personalSteps.legend')}
                />
            </Card>
            <Card
                title={t('personalSteps.added')}
                subtitle={t('personalSteps.addedSubtitle')}
            >
                <StepsChart
                    query={GET_STAGES_REPORT_STEPS_ADDED}
                    mapData={data =>
                        data.organizationPathwayStagesReport.map(row => ({
                            [t('personalSteps.label')]: row.stepsAddedCount,
                            [t('stage')]: row.pathwayStage.name.toUpperCase(),
                        }))
                    }
                    label={t('personalSteps.label')}
                    index={t('stage')}
                    variables={{
                        period: '',
                        organizationId: orgId,
                        endDate: moment().format(),
                    }}
                />
            </Card>
            <ImpactInfo
                query={GET_IMPACT_REPORT_MOVED}
                text={report =>
                    t('personalSteps.reached', {
                        count: report.impactReport.pathwayMovedCount,
                    })
                }
                variables={{ organizationId: orgId }}
            />
            <Card
                title={t('personalSteps.members')}
                subtitle={t('personalSteps.membersSubtitle')}
            >
                <StepsChart
                    query={GET_STAGES_REPORT_MEMBER_COUNT}
                    mapData={data =>
                        data.organizationPathwayStagesReport.map(row => ({
                            [t('members')]: row.memberCount,
                            [t('stage')]: row.pathwayStage.name.toUpperCase(),
                        }))
                    }
                    label={t('members')}
                    index={t('stage')}
                    variables={{
                        period: '',
                        organizationId: orgId,
                        endDate: moment().format(),
                    }}
                />
            </Card>
        </div>
    );
};

export default PersonalStepsPage;

PersonalStepsPage.propTypes = {};
