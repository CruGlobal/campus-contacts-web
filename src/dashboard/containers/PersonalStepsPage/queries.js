import gql from 'graphql-tag';

const GET_IMPACT_REPORT_MOVED = gql`
    query impactReport($organizationId: ID!) {
        impactReport(organizationId: $organizationId) {
            stageProgressionCount
        }
    }
`;

const GET_IMPACT_REPORT_STEPS_TAKEN = gql`
    query impactReport($organizationId: ID!) {
        impactReport(organizationId: $organizationId) {
            personalStepsCount
        }
    }
`;

const GET_STAGES_REPORT_MEMBER_COUNT = gql`
    query organizationStagesReport(
        $period: String!
        $organizationId: ID!
        $endDate: ISO8601DateTime!
    ) {
        organizationStagesReport(
            period: $period
            organizationId: $organizationId
            endDate: $endDate
        ) {
            memberCount
            stage {
                name
            }
        }
    }
`;

const GET_STAGES_REPORT_STEPS_ADDED = gql`
    query organizationStagesReport(
        $period: String!
        $organizationId: ID!
        $endDate: ISO8601DateTime!
    ) {
        organizationStagesReport(
            period: $period
            organizationId: $organizationId
            endDate: $endDate
        ) {
            stage {
                name
            }
            personalStepsAddedCount
        }
    }
`;

const GET_STEPS_COMPLETED_REPORT = gql`
    query communityReport {
        communityReport {
            dayReport {
                date
                personalStepsCompletedCount
                communityStagesReport {
                    pathwayStage
                    personalStepsCompletedCount
                }
            }
        }
    }
`;

const GET_TOTAL_STEPS_COMPLETED_SUMMARY = gql`
    query communityReport {
        communityReport {
            communityStagesReport {
                pathwayStage
                personalStepsCompletedCount
            }
        }
    }
`;

export {
    GET_IMPACT_REPORT_MOVED,
    GET_IMPACT_REPORT_STEPS_TAKEN,
    GET_STAGES_REPORT_MEMBER_COUNT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT_STEPS_ADDED,
    GET_TOTAL_STEPS_COMPLETED_SUMMARY,
};
