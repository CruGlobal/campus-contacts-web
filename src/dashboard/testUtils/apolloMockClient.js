import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
import { buildClientSchema } from 'graphql/utilities';
import { createMockerFromIntrospection } from 'fraql/mock';
import gql from 'fraql';

import introspectionQuery from '../../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument, options) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
