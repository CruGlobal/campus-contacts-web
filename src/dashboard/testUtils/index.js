import React, { ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';

import { createApolloMockClient } from './apolloMockClient';
import defaultTheme from '../defaultTheme';
import { ThemeProvider } from 'emotion-theming';
import AppContext from '../appContext';

export const renderWithContext = (component, { mocks } = {}, context = {}) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={context}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });

    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
