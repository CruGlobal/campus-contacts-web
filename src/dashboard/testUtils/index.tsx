import React, { ReactNode } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';

import defaultTheme from '../defaultTheme';
import AppContext from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

export const renderWithContext = (
    component: any,
    { mocks: mocks = {}, appContext = {} } = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    // @ts-ignore
    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
