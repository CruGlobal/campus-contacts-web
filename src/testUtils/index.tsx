import React, { ReactNode, ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import { IMocks } from 'graphql-tools';

import defaultTheme from '../defaultTheme';
import {
    AppContext,
    AppContextValue,
    appContextDefaultValue,
} from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

interface RenderWithContextParams {
    mocks?: IMocks;
    appContext?: AppContextValue;
}

export const renderWithContext = (
    component: ReactElement,
    {
        mocks: mocks = {},
        appContext = appContextDefaultValue,
    }: RenderWithContextParams = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children?: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
