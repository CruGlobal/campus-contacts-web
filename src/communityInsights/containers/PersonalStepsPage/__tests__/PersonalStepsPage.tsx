import React from 'react';

import { renderWithContext } from '../../../../testUtils';

import PersonalStepsPage from '..';

describe('<PersonalStepsPage />', () => {
    it('should render properly in loading state', async () => {
        const { snapshot, unmount } = renderWithContext(<PersonalStepsPage />, {
            appContext: {
                orgId: '1',
            },
        });
        snapshot();
        unmount();
    });

    // TODO: Add a test with mocking data when GraphQL schema contains above query
    /*
    it('should render properly with chart', async () => {
        const { snapshot, getByText } = renderWithContext(
            <PersonalStepsPage />,
            {
                mocks: {
                    Query: () => ({
                        communityReport: () => ({
                            communityStagesReport: () => []
                        }),
                        organizationPathwayStagesReport: () => [
                            {
                                stepsAddedCount: 20,
                                memberCount: 10,
                                pathwayStage: {
                                    name: 'NO STAGE',
                                },
                            },
                            {
                                stepsAddedCount: 25,
                                memberCount: 14,
                                pathwayStage: {
                                    name: 'UNINTERESTED',
                                },
                            },
                        ],
                        impactReport: () => ({
                            pathwayMovedCount: () => 13,
                            stepOwnersCount: () => 16
                        }),
                    }),
                },
                appContext: {
                    orgId: 1,
                },
            },
        );

        await waitForElement(() => getByText('NO STAGE'));
        snapshot();
    });
    */
});
