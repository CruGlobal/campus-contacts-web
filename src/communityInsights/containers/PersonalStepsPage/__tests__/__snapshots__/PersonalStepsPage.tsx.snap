// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalStepsPage /> should render properly in loading state 1`] = `
.emotion-0 {
  font-weight: 300;
  font-size: 30px;
  line-height: 38px;
  color: #505256;
  margin-bottom: 0px;
  margin-top: 48px;
  padding-left: 21px;
}

.emotion-33 {
  box-shadow: 0px 2px 12px rgba(0,0,0,0.2);
  background: #ffffff;
  border-radius: 8px;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 43px;
  margin-top: 36px;
}

.emotion-1 {
  font-weight: 300;
  font-size: 36px;
  line-height: 38px;
  color: #505256;
  margin-bottom: 12px;
  margin-top: 0px;
}

.emotion-2 {
  font-size: 14px;
  line-height: 20px;
  font-weight: normal;
  color: #505256;
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-32 {
  padding: 25px;
  margin: 0;
}

.emotion-31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  height: 52px;
  width: 52px;
  background-size: contain;
  background-repeat: no-repeat;
}

.emotion-3.not-sure {
  background-image: url(mockFile);
}

.emotion-3.uninterested {
  background-image: url(mockFile);
}

.emotion-3.curious {
  background-image: url(mockFile);
}

.emotion-3.forgiven {
  background-image: url(mockFile);
}

.emotion-3.growing {
  background-image: url(mockFile);
}

.emotion-3.guiding {
  background-image: url(mockFile);
}

.emotion-3.discipleship-conversation {
  background-image: url(mockFile);
}

.emotion-3.personal-evangelism {
  background-image: url(mockFile);
}

.emotion-3.holy-spirit-presentation {
  background-image: url(mockFile);
}

.emotion-3.personal-evangelism-decisions {
  background-image: url(mockFile);
}

.emotion-3.spiritual-conversation {
  background-image: url(mockFile);
}

.emotion-4 {
  font-weight: 600;
  font-size: 9px;
  line-height: 14px;
  color: #B4B6BA;
  margin-top: 17px;
  text-transform: uppercase;
  overflow-wrap: break-word;
  width: 110px;
  text-align: center;
  height: 14px;
}

.emotion-5 {
  font-weight: 600;
  font-size: 32px;
  line-height: 38px;
  color: #007398;
  margin-top: 9px;
}

.emotion-30 {
  margin-top: 28px;
}

.emotion-29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-29 .DateRangePickerInput {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-29 .DateInput {
  width: 108px;
  height: 32px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #eceef2;
  text-align: center;
  border-radius: 0 16px 16px 0;
  overflow: hidden;
  cursor: pointer;
}

.emotion-29 .DateInput:first-of-type {
  border-radius: 16px 0 0 16px;
  margin-right: 1px;
}

.emotion-29 .DateInput:hover {
  background-color: #e2e2e2;
}

.emotion-29 .DateInput input[type='text']:focus {
  background-color: #e2e2e2;
}

.emotion-29 .DateInput_input {
  height: 100%;
  font-size: 14px;
  line-height: 20px;
  padding: 0;
  background-color: transparent;
  border-color: transparent;
  border-radius: 0;
  text-align: center;
  color: #505256;
  cursor: pointer;
}

.emotion-29 .CalendarDay__selected_span {
  background: #3CC8E6;
  color: white;
  border: 1px solid #2fb0d8;
}

.emotion-29 .CalendarDay__selected {
  background: #007398;
  color: white;
  border: 1px solid #2fb0d8;
}

.emotion-29 .CalendarDay__selected:hover {
  background: #007398;
  color: white;
}

.emotion-29 .CalendarDay__hovered_span:hover,
.emotion-29 .CalendarDay__hovered_span {
  background: #3CC8E6;
  border: 1px solid #2fb0d8;
  color: white;
}

.emotion-29 .DateRangePickerInput_arrow {
  display: none;
}

.emotion-29 .DateInput_fang {
  top: 56px;
}

.emotion-28 {
  background-image: url(mockFile);
  width: 24px;
  height: 24px;
  margin-right: 12px;
}

.emotion-37 {
  width: 100%;
  height: 190px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
  border-radius: 8px;
}

.emotion-36 {
  text-align: center;
  width: 720px;
  color: #007398;
  font-size: 16px;
  font-weight: 300;
  line-height: 24px;
  margin: 10px 0;
}

<div>
  <div>
    <h1
      class="emotion-0"
    >
      <span>
        Gathering your stats...
      </span>
    </h1>
    <div
      class="emotion-33"
    >
      <h1
        class="emotion-1"
      >
        Total Personal Steps of Faith Completed
      </h1>
      <h2
        class="emotion-2"
      />
      <div
        class="emotion-32"
      >
        <div
          class="emotion-31"
        >
          <div
            class="emotion-27"
          >
            <div
              class="emotion-6"
            >
              <div
                class="uninterested emotion-3"
              />
              <div
                class="emotion-4"
              >
                Uninterested
              </div>
              <div
                class="emotion-5"
              >
                -
              </div>
            </div>
            <div
              class="emotion-6"
            >
              <div
                class="curious emotion-3"
              />
              <div
                class="emotion-4"
              >
                Curious
              </div>
              <div
                class="emotion-5"
              >
                -
              </div>
            </div>
            <div
              class="emotion-6"
            >
              <div
                class="forgiven emotion-3"
              />
              <div
                class="emotion-4"
              >
                Forgiven
              </div>
              <div
                class="emotion-5"
              >
                -
              </div>
            </div>
            <div
              class="emotion-6"
            >
              <div
                class="growing emotion-3"
              />
              <div
                class="emotion-4"
              >
                Growing
              </div>
              <div
                class="emotion-5"
              >
                -
              </div>
            </div>
            <div
              class="emotion-6"
            >
              <div
                class="guiding emotion-3"
              />
              <div
                class="emotion-4"
              >
                Guiding
              </div>
              <div
                class="emotion-5"
              >
                -
              </div>
            </div>
            <div
              class="emotion-6"
            >
              <div
                class="not-sure emotion-3"
              />
              <div
                class="emotion-4"
              >
                Not Sure
              </div>
              <div
                class="emotion-5"
              >
                -
              </div>
            </div>
          </div>
          <div
            class="emotion-30"
          >
            <div
              class="emotion-29"
            >
              <div
                class="emotion-28"
              />
              <div
                class="DateRangePicker DateRangePicker_1"
              >
                <div>
                  <div
                    class="DateRangePickerInput DateRangePickerInput_1"
                  >
                    <div
                      class="DateInput DateInput_1"
                    >
                      <input
                        aria-describedby="DateInput__screen-reader-message-startDateId"
                        aria-label="Start Date"
                        autocomplete="off"
                        class="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
                        id="startDateId"
                        name="startDateId"
                        placeholder="Start Date"
                        readonly=""
                        type="text"
                        value="09/25/2019"
                      />
                      <p
                        class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                        id="DateInput__screen-reader-message-startDateId"
                      >
                        Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                      </p>
                    </div>
                    <div
                      aria-hidden="true"
                      class="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
                      role="presentation"
                    >
                      <svg
                        class="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
                        focusable="false"
                        viewBox="0 0 1000 1000"
                      >
                        <path
                          d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
                        />
                      </svg>
                    </div>
                    <div
                      class="DateInput DateInput_1"
                    >
                      <input
                        aria-describedby="DateInput__screen-reader-message-endDateId"
                        aria-label="End Date"
                        autocomplete="off"
                        class="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
                        id="endDateId"
                        name="endDateId"
                        placeholder="End Date"
                        readonly=""
                        type="text"
                        value="10/02/2019"
                      />
                      <p
                        class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                        id="DateInput__screen-reader-message-endDateId"
                      >
                        Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-33"
    >
      <h1
        class="emotion-1"
      >
        Personal Steps of Faith Completed
      </h1>
      <h2
        class="emotion-2"
      >
        Personal steps of faith completed by community members.
      </h2>
      <div
        class="emotion-32"
      >
        <div
          class="emotion-37"
        >
          <img
            src="mockFile"
          />
          <p
            class="emotion-36"
          >
            This community does not have any members who have completed a personal step of faith. Challenge your community to open the MissionHub mobile app and complete a personal step of faith.
          </p>
        </div>
      </div>
    </div>
    <div
      class="emotion-33"
    >
      <h1
        class="emotion-1"
      >
        Personal Steps of Faith Added
      </h1>
      <h2
        class="emotion-2"
      >
        Current total of personal steps of faith added by members by stage.
      </h2>
      <div
        class="emotion-32"
      >
        <div
          class="emotion-37"
        >
          <img
            src="mockFile"
          />
          <p
            class="emotion-36"
          >
            This community does not have any members with personal steps of faith. Challenge your community to open the MissionHub mobile app and add a personal step of faith.
          </p>
        </div>
      </div>
    </div>
    <h1
      class="emotion-0"
    >
      <span>
        Gathering your stats...
      </span>
    </h1>
    <div
      class="emotion-33"
    >
      <h1
        class="emotion-1"
      >
        Community Member Stages
      </h1>
      <h2
        class="emotion-2"
      >
        Current total number of members by stage.
      </h2>
      <div
        class="emotion-32"
      >
        <div
          class="emotion-37"
        >
          <img
            src="mockFile"
          />
          <p
            class="emotion-36"
          >
            This community does not have any members with stages. Challenge your current members to open the MissionHub mobile app and choose a stage.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
